dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/Governors/Standard/Steam/IEEEG1.mo

```diff
@@ -36,82 +34,68 @@ model IEEEG1 "Steam turbine governor"
     Placement(visible = true, transformation(origin = {-50, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
-  Modelica.Blocks.Nonlinear.Limiter limiter(uMax = DerPMaxPu, uMin = DerPMinPu) annotation(
+  Modelica.Blocks.Nonlinear.Limiter limiter(uMax = Uo, uMin = Uc) annotation(
     Placement(visible = true, transformation(origin = {-90, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
-  Modelica.Blocks.Continuous.FirstOrder firstOrder3(T = t7, initType = Modelica.Blocks.Types.Init.InitialState, y_start = PBoilerPu) annotation(
+  Modelica.Blocks.Continuous.FirstOrder firstOrder3(T = t7, y_start = PmRef0Pu) annotation(
     Placement(visible = true, transformation(origin = {190, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
   Modelica.Blocks.Continuous.TransferFunction transferFunction(a = {t1, 1}, b = {t2, 1}) annotation(
     Placement(visible = true, transformation(origin = {-230, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
   Modelica.Blocks.Math.Sum sum1(k = {K1, K3, K5, K7}, nin = 4) annotation(
     Placement(visible = true, transformation(origin = {250, 100}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
   Modelica.Blocks.Math.Sum sum2(k = {K2, K4, K6, K8}, nin = 4) annotation(
     Placement(visible = true, transformation(origin = {250, -100}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
-  Modelica.Blocks.Math.Gain gain1(k = SNom / PNomTurb) annotation(
-    Placement(visible = true, transformation(origin = {290, 100}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
-  Modelica.Blocks.Math.Gain gain2(k = SNom / PNomTurb) annotation(
-    Placement(visible = true, transformation(origin = {290, -100}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Add add(k2 = -1) annotation(
+    Placement(visible = true, transformation(origin = {-270, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Add add1 annotation(
+    Placement(visible = true, transformation(origin = {310, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Add3 add3(k2 = -K, k3 = -1) annotation(
+    Placement(visible = true, transformation(origin = {-170, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
 
+  //Generator initial parameter
   parameter Types.ActivePowerPu Pm0Pu "Initial mechanical power in pu (base PNomTurb)";
 
-  final parameter Types.ActivePowerPu PBoilerPu = if K1 + K3 + K5 + K7 > 0 then Pm0Pu * (PNomTurb / SNom) / (K1 + K3 + K5 + K7) else 0 "Initial power output of boiler in pu (base SNom)";
-  final parameter Types.ActivePowerPu PmRef0Pu = PBoilerPu + K * (SystemBase.omega0Pu - SystemBase.omegaRef0Pu) "Initial reference mechanical power in pu (base SNom)";
+  final parameter Types.ActivePowerPu PmRef0Pu = Pm0Pu / (K1 + K2 + K3 + K4 + K5 + K6 + K7 + K8) "Initial reference mechanical power in pu (base PNomTurb)";
```

Comment:
We can put them to have a more generic model

Reply:
As you wish.

The assumption SystemBase.omega0Pu = SystemBase.omegaRef0Pu is widespread throughout the library, just check HyGov, Gast, TGov1, TGov3.

Reply:
Moreover, if we are to account for omega0Pu - omegaRef0Pu in PmRefPu, we ought to add to the transferFunction block a start value : x_start = SystemBase.omega0Pu - SystemBase.omegaRef0Pu, x_scaled(start =  SystemBase.omega0Pu - SystemBase.omegaRef0Pu)

Reply:
In IEEEG11.mo (formerly IEEEG1.mo), omegaRefPu is replaced by a constant equal to omega0Pu. The transferFunction block has a (default) zero start value, this is consistent. But PmRef0Pu = PBoiler0Pu + K * (omega0Pu - omegaRef0Pu), an absurdity since omegaRefPu appears nowhere.

I suggest to keep the assumption that omega0Pu = omegaRef0Pu and thus to leave the code as it is. We might open a ticket to allow for omega0Pu <> omegaRef0Pu (thus turning these constants into parameters), would it be useful ?

Reply:
No need to create another ticket for now.